rotkehlchen/globaldb/handler.py

```diff
@@ -1347,6 +1349,36 @@ def soft_reset_assets_list() -> Tuple[bool, str]:
         cursor.execute(detach_database)
         return True, ''
 
+    def save_binance_pairs(
+        self,
+        new_pairs: Iterable['BinancePair'],
+        location: Location,
+    ) -> bool:
+        query = 'INSERT OR IGNORE INTO binance_pairs(pair, base_asset, quote_asset, location) VALUES (?, ?, ?, ?)'  # noqa: E501
+        cursor = self._conn.cursor()
+        try:
+            cursor.executemany(query, [pair.serialize_for_db() for pair in new_pairs])
+            self.add_setting_value(name=f'binance_pairs_queried_at_{location}', value=ts_now())
+            return True
+        except sqlite3.IntegrityError as e:
+            log.error(f'Failed to store binance pairs in database. {str(e)}')
```

Comment:
This is a bad error. Don't just log and return False. Raise an `InputError` or something close to it and catch it at api level and return an error to the user.

From what I see you don't; even check the return of this function but just happily continue :sweat: